pkg/i2gw/providers/ingressnginx/ingressnginx.go

```diff
@@ -32,14 +32,17 @@ type Provider struct {
 	conf *i2gw.ProviderConf
 
 	*resourceReader
+	*resourceFilter
 	*converter
 }
 
 // NewProvider constructs and returns the ingress-nginx implementation of i2gw.Provider.
 func NewProvider(conf *i2gw.ProviderConf) i2gw.Provider {
+	conf.FilteredObjects = filteredObjects
```

Comment:
If I'm not mistaken, all providers share the same ProviderConf object. So, different providers will override the `FilteredObjects` with their own filteredObjects list, meaning the last constructed provider's filtered objects will last

Reply:
Nice catch @levikobi! I've changed, now the configuration is passed by value and the objects are not overwritten